Guide for input/output in original FW. (#20141)
openvinotoolkit/openvino: docs/OV_Converter_UG/prepare_model/convert_model/MO_OVC_transition.md

            - Not available in OVC tool. Please check Python API.

> Input and output cut

Reviewer: ```suggestion
Cutting Off Parts of a Model
```

##########################

> In new conversion API input and output model cut are no longer available. It is recommended to perform cut in original framework instead.

Reviewer: ```suggestion
In new conversion API input and output model cutting is no longer available. It is recommended to perform cutting in original framework instead.
```

> Below are examples of input and output cut for TF1, TF saved model and ONNX formats in legacy conversion API with example of how to perform identical cut in original framework.

Reviewer: ```suggestion
Below are examples of input and output cutting for TensorFlow protobuf, TensorFlow SavedModel and ONNX formats in legacy conversion API with example of how to perform identical cut in original framework.
```

> For PyTorch, TF Keras and PaddlePaddle it is recommended to change original model code to perform model cut. 

Reviewer: ```suggestion
For PyTorch, TensorFlow 2 Keras and PaddlePaddle it is recommended to change original model code to perform model cut. 
```

> ``TF frozen .pb format / frozen ft.Graph / tf.GraphDef``

Reviewer: ```suggestion
``TensorFlow protobuf format / tf.Graph / tf.GraphDef``
```

                      return graph
               
               
              # Load TF model
              graph = load_graph("/path_to_model/HugeCTR.pb")
               
              # Convert the model with input and output cut
              input = "concat"
              output = "MatVec_3/Squeeze"
              ov_model = mo.convert_model(graph, input=(input, [-1, -1]), output=output)
               
              # Compile the model
              compiled_model = ov.compile_model(ov_model)

         - .. code-block:: py
              :force:

              import openvino as ov
              import tensorflow as tf
               
              from tensorflow.python.tools.strip_unused_lib import strip_unused
               
              def load_graph(model_path):
                  graph_def = tf.compat.v1.GraphDef()
                  with open(model_path, "rb") as f:
                      graph_def.ParseFromString(f.read())
                  with tf.compat.v1.Graph().as_default() as graph:
                      tf.graph_util.import_graph_def(graph_def, name="")
                      return graph
               
              # Load TF model

>               graph = load_graph("/home/panas/HugeCTR.pb")

Reviewer: ```suggestion
              graph = load_graph("HugeCTR.pb")
```

              
              # Cut the model
              input = "concat"

>               output = "MatVec_3/Squeeze"

Reviewer: I think that cutting should be by tensors so it should contain `:0`

Author: `strip_unused()` works only with node names. Adding `:0` leads to error:
ValueError: Name 'concat:0' appears to refer to a Tensor, not an Operation.

              graph_def = graph.as_graph_def()
              new_graph_def = strip_unused(graph_def, [input], [output], tf.float32.as_datatype_enum)
               
              # Convert and compile model
              ov_model = ov.convert_model(new_graph_def, input=[-1, -1])
              cmp_model = ov.compile_model(ov_model)

> ``TF Saved Model format / non-frozen ft.Graph``

Reviewer: ```suggestion
``TensorFlow SavedModel format``
```

###############################################

.. tab-set::
    .. list-table::
       :header-rows: 1

       * - Legacy API
         - New API
       * - .. Not available in legacy API

         - .. code-block:: py
              :force:

>               import numpy as np

Reviewer: not used

>               import numpy as np
> 

Reviewer: ```suggestion
```

Reviewer: looks not used

              from tensorflow.python.tools.strip_unused_lib import strip_unused
               
               
              # Load TF model
              model = hub.load("https://tfhub.dev/svampeatlas/vision/embedder/fungi_V2/1?tf-hub-format=compressed")
               
              # Convert model to GraphDef
              model_func = model.signatures["default"]
              frozen_func = convert_variables_to_constants_v2(model_func)
              graph_def = frozen_func.graph.as_graph_def()
               
              # Cut the model
              input_name = 'InceptionV4/InceptionV4/Conv2d_2b_3x3/Relu'
              output_name = 'InceptionV4/InceptionV4/Mixed_7c/concat'
              new_graph_def = strip_unused(graph_def, [input_name], [output_name], tf.float32.as_datatype_enum)
               
              # Convert and compile the model
              ov_model = ov.convert_model(new_graph_def)
              compiled_model = ov.compile_model(ov_model)

``ONNX``
########

.. tab-set::
    .. list-table::
       :header-rows: 1

       * - Legacy API
         - New API
       * - .. code-block:: py
              :force:

>               import onnx

Reviewer: onnx is not used

Author: Thanks! Removed.

              import openvino as ov
              import openvino.tools.mo as mo
               

>               input_path = "/path_to_model/yolov8x.onnx"

Reviewer: ```suggestion
              input_path = "yolov8x.onnx"
```

               
              # Convert model and perform input and output cut
              input = "/model.2/Concat_output_0"
              output = "/model.22/Concat_3_output_0"
              ov_model = mo.convert_model(input_path, input=input, output=output)
               
              #Compile model
              ov.compile_model(ov_model)

         - .. code-block:: py
              :force:

              import onnx
              import openvino as ov
               

>               input_path = "/path_to_model/yolov8x.onnx"

Reviewer: ```suggestion
              input_path = "yolov8x.onnx"
```
